File: Tugas Kuis Peluang/bagian1_2.py
import math
import os

def clear_terminal():
    os.system("cls")

def prob_mobil(t, p):
    # mencari laju kejadian per menit
    lajuKejadian = -math.log(1-p)/t

    # mencari peluang melihat mobil dalam 10 menit
    waktu = 10 / 60 # konversi menit ke jam
    prob = 1 - math.exp(-lajuKejadian * waktu)
    return prob

def konversiPersen(prob_10):
    persen = prob_10 * 100
    return "{:.2f}%".format(persen)

clear_terminal()
# Meminta input dari user
p = float(input("Masukkan nilai peluang melihat mobil dalam 30 menit: "))
t = float(input("Masukkan waktu dalam satuan menit: "))

prob_10 = prob_mobil(t, p)
persentase_10 = konversiPersen(prob_10)

print("\n")
print("Peluang melihat mobil di jalan raya dalam 10 menit adalah : ", prob_10)
print("Persentase melihat mobil di jalan raya dalam 10 menit adalah : ", persentase_10)

File: Tugas Kuis Peluang/bagian1_2.c
#include <stdio.h>
#include <math.h>

int main(){
    // deklarasi variabel
    int jumlah_mobil = 1; // asumsikan melihat 1 mobil
    float prob_mobil;
    int waktu;
    double lambda, peluang, persentase;

    // input
    system("cls");
    printf("Masukkan nilai peluang melihat mobil dalam 30 menit: ");
    scanf("%f", &prob_mobil);
    printf("Masukkan waktu dalam satuan menit: ");
    scanf("%d", &waktu);

    // proses
    lambda = prob_mobil * waktu / 30;
    peluang = exp(-lambda) * pow(lambda, jumlah_mobil) / tgamma(jumlah_mobil+1); // menghitung peluang dengan distribusi poison
    persentase = peluang * 100; // menghitung persentase

    // output
    printf("\nPeluang melihat %d mobil dalam %d menit: %.4f", jumlah_mobil, waktu, peluang);
    printf("\nPersentase melihat %d mobil dalam %d menit: %.2f%%", jumlah_mobil, waktu, persentase);

    return 0;
}
